### ISSUE TITLE
Multistep weights in forecast loss not applied

### ISSUE THREAD

**Issue by robert-DL** [INRIA/geoarches#36]
Hi Folks, 

The issue is about the code snippet here:
https://github.com/INRIA/geoarches/blob/929d6b2364fa29dc14fdd0c6ac80cae37ef293e5/geoarches/lightning_modules/forecast.py#L166C13-L166C59

td.apply()  is not in-place!(https://docs.pytorch.org/tensordict/stable/reference/generated/tensordict.TensorDict.html#tensordict.TensorDict.apply)
The desired in-place operation is `td.apply_()`.
I therefore expect that the multistep weights are not applied at all (but I did not test for it). I just checked the tensordict documentation and tried a dummy example. In the example my hypothesis holds true. 

Best 
Robert

**Comment by 14renus** [comment]
Indeed this is something we should fix!
However, this should not affect any results in the paper since ArchesWeatherGen was not trained with multistep loss.
